c++/src/kj/async-io-test.c++

```diff
@@ -3574,5 +3574,203 @@ KJ_TEST("pump file to socket") {
   doTest(fs->getCurrent().createTemporary());
 }
 
+// ---------------------------------------------------------------------------------------------
+// accept() with aborted connection - IPv4 listener
+// ---------------------------------------------------------------------------------------------
+// Test accept() behavior when connections are aborted (send RST) before being accepted.
+// Creates an aborted IPv4 connection followed by a valid one, then verifies proper handling.
+//
+// On Unix platforms, accept() typically returns the aborted connection, which fails on first
+// read (throws "Connection reset by peer" or returns 0 bytes). The test then calls accept()
+// again to get the valid connection. Some platforms may filter out aborted connections.
+
+#if !_WIN32
+KJ_TEST("accept() with aborted connection - IPv4") {
+  // -- async-io boilerplate ---------------------------------------------------
+  auto io = kj::setupAsyncIo();
+  auto& net = io.provider->getNetwork();
+
+  auto listenerAddr = net.parseAddress("127.0.0.1", 0).wait(io.waitScope);
+  auto listener = listenerAddr->listen();
+  uint16_t port = listener->getPort();
+
+  // Create a connection that will be aborted (sends RST packet)
+  kj::Thread abortThread([&] {
+    int s = ::socket(AF_INET, SOCK_STREAM, 0);
+    KJ_ASSERT(s >= 0);
+
+    // Configure socket to send RST on close instead of FIN
+    struct linger lg = {1, 0};
+    KJ_SYSCALL(setsockopt(s, SOL_SOCKET, SO_LINGER, &lg, sizeof(lg)));
+
+    sockaddr_in a {};
+    a.sin_family = AF_INET;
+    a.sin_port = htons(port);
+    a.sin_addr.s_addr = htonl(0x7f000001);
+    KJ_SYSCALL(::connect(s, reinterpret_cast<sockaddr*>(&a), sizeof(a)));
+
+    // Close immediately to send RST packet
+    KJ_SYSCALL(::close(s));
+  });
+
+  // Allow aborted connection to reach accept queue first
+  io.provider->getTimer().afterDelay(2 * kj::MILLISECONDS)
+                        .wait(io.waitScope);
+
+  // Create a valid connection that should work
+  auto clientP = net.parseAddress("127.0.0.1", port)
+      .then([](Own<NetworkAddress> addr) { return addr->connect(); });
+  // Accept first connection - this should be the aborted one
+  auto firstConnection = listener->accept().wait(io.waitScope);
+  KJ_ASSERT(firstConnection);
```

Comment:
This is redundant, `Own<T>` should never be null (otherwise it should be `Maybe<Own<T>>`). I'm actually slightly surprised this compiles.

```suggestion
```